L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -0,0 +1,462 @@
+#include "L1Trigger/L1TMuonCPPF/interface/RecHitProcessor.h"
+
+RecHitProcessor::RecHitProcessor() {
+}
+
+RecHitProcessor::~RecHitProcessor() {
+}
+
+void RecHitProcessor::processLook(
+				  const edm::Event& iEvent,
+				  const edm::EventSetup& iSetup,
+				  const edm::EDGetToken& recHitToken,
+				  std::vector<RecHitProcessor::CppfItem>& CppfVec1,
+				  l1t::CPPFDigiCollection& cppfDigis
+				  ) const {
+  
+  edm::Handle<RPCRecHitCollection> recHits;
+  iEvent.getByToken(recHitToken, recHits);
+  
+  edm::ESHandle<RPCGeometry> rpcGeom;
+  iSetup.get<MuonGeometryRecord>().get(rpcGeom);
+  
+  // Loop over ... what, exactly?
+  for ( TrackingGeometry::DetContainer::const_iterator iDet = rpcGeom->dets().begin(); iDet < rpcGeom->dets().end(); iDet++ ) {
+    
+    // What does this do?  Check whether the chamber is an RPC chamber?
+    if (dynamic_cast<const RPCChamber*>( *iDet ) == nullptr ) continue;
+    
+    auto chamb = dynamic_cast<const RPCChamber* >( *iDet ); 
+    std::vector<const RPCRoll*> rolls = (chamb->rolls());
+    
+    // Loop over rolls in the chamber
+    for (std::vector<const RPCRoll*>::const_iterator iRoll = rolls.begin(); iRoll != rolls.end(); ++iRoll) {
+      
+      RPCDetId rpcId = (*iRoll)->id();	
+      
+      typedef std::pair<RPCRecHitCollection::const_iterator, RPCRecHitCollection::const_iterator> rangeRecHits;
+      rangeRecHits recHitCollection =  recHits->get(rpcId);
+      
+      
+      
+      //Loop over the RPC digis
+      for (RPCRecHitCollection::const_iterator rechit_it = recHitCollection.first; rechit_it != recHitCollection.second; rechit_it++) {	
+	
+	//const RPCDetId& rpcId = rechit_it->rpcId();
+	int rawId = rpcId.rawId();
+	//int station = rpcId.station();
+	int Bx = rechit_it->BunchX(); 
+	int isValid = rechit_it->isValid();
+	int firststrip = rechit_it->firstClusterStrip();
+	int clustersize = rechit_it->clusterSize();
+	LocalPoint lPos = rechit_it->localPosition();
+	const RPCRoll* roll = rpcGeom->roll(rpcId);
+	const BoundPlane& rollSurface = roll->surface();
+	GlobalPoint gPos = rollSurface.toGlobal(lPos);
+	float global_theta = emtf::rad_to_deg(gPos.theta().value());
+	float global_phi   = emtf::rad_to_deg(gPos.phi().value());
+	//::::::::::::::::::::::::::::
+	//Establish the average position of the rechit    
+	int rechitstrip = firststrip;
+	
+        if(clustersize > 2) {
+	  int medium = 0;
+	  if (clustersize % 2 == 0) medium = 0.5*(clustersize); 
+	  else medium = 0.5*(clustersize-1);
+	  rechitstrip += medium; 
+	} 
+	
+	if(clustersize > 3) continue;	
+	//This is just for test CPPFDigis with the RPC Geometry, It must be "False" in the normal runs 
```

Comment:
It should be set to false in that case. Maybe use a configuration parameter instead for easier handling.